src/main/java/org/takes/facets/auth/social/PsGithub.java

```diff
@@ -153,19 +153,25 @@ private String token(final String home, final String code)
         final String uri = new Href(this.github)
             .path("login").path("oauth").path("access_token")
             .toString();
-        return new JdkRequest(uri)
-            .method("POST")
-            .header("Accept", "application/xml")
-            .body()
-            .formParam("client_id", this.app)
-            .formParam("redirect_uri", home)
-            .formParam("client_secret", this.key)
-            .formParam("code", code)
-            .back()
-            .fetch().as(RestResponse.class)
-            .assertStatus(HttpURLConnection.HTTP_OK)
-            .as(XmlResponse.class)
-            .xml().xpath("/OAuth/access_token/text()").get(0);
+        try {
+            return new JdkRequest(uri)
+                .method("POST")
+                .header("Accept", "application/xml")
+                .body()
+                .formParam("client_id", this.app)
+                .formParam("redirect_uri", home)
+                .formParam("client_secret", this.key)
+                .formParam("code", code)
+                .back()
+                .fetch().as(RestResponse.class)
+                .assertStatus(HttpURLConnection.HTTP_OK)
+                .as(XmlResponse.class)
+                .xml().xpath("/OAuth/access_token/text()").get(0);
+        } catch (final IndexOutOfBoundsException exception) {
```

Comment:
@yegor256 The problem is the call to xpath is throwing the exception, so it won't reach the tokens.isEmpty line.

- I would like to have the return statement in the last line of the method, but @pinaf suggested this is the better way